### PR TITLE
Data model persistence

### LINKED ISSUE #31: Add data model persistence

### LINKED ISSUE #62: Convert MessagesManager to a singleton (to follow the same pattern as the other Managers)

### PR DESCRIPTION
**Pull request by maxwellpark**
Make sure you take out life insurance before reviewing these 65 files.

### CODE REVIEW

**Comment by maxwellpark** [comment]
It only supports one player save file at the moment. 

Some stuff that didn't need to be committed but after a while I couldn't be bothered to uncheck files.

File: Assets/Scripts/DataModels/DataModelsUtils.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    {
+        Debug.Log($"{typeof(T)} load commencing.");
+        string folderPath = GetSaveFolderPath(folderName);
+        string filePath = Path.Combine(folderPath, fileName + FILE_EXTENSION);
+
+        if (File.Exists(filePath))
+        {
+            try
+            {
+                string json;
```
Comment (maxwellpark) [inline Assets/Scripts/DataModels/DataModelsUtils.cs:57]: I was getting json parse errors invalid value with the code we did last time. 
This seems to work fine. And it is less lines so why not. I don't think there's any downside.

Comment (ambid17) [reply Assets/Scripts/DataModels/DataModelsUtils.cs:57]: yeah this should be fine

File: Assets/Scripts/DataModels/Messages/MessageAccessors.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿public partial class Message
+{
```
Comment (ambid17) [inline Assets/Scripts/DataModels/Messages/MessageAccessors.cs:2]: OOOOOH CLEEAANNN

File: Assets/Scripts/DataModels/Missions/MissionOutcomes/PilotXpOutcome.cs [hunk 3]
```diff
@@ -1,29 +1,29 @@
...
+	}
+
+    // A mission can have multiple omens attached to it
+    private float ApplyOmens(Mission mission)
+    {
+        float coefficient = 1f;
+
+        for (int i = 0; i < mission.Outcomes.Length; i++)
+        {
+            if (mission.Outcomes[i].GetType().Equals(typeof(OmenOutcome)))
```
Comment (ambid17) [inline Assets/Scripts/DataModels/Missions/MissionOutcomes/PilotXpOutcome.cs]: cant you just do `mission.Outcomes[i] is OmenOutcome`

**Review by ambid17 — APPROVED** [review]

**Pull request merged**